browser-extension/mv2/src/client/js/client.js

```diff
@@ -3,25 +3,14 @@
     return;
   }
 
-  chrome.runtime.sendMessage(
-    {
-      action: RQ.EXTENSION_MESSAGES.CHECK_IF_EXTENSION_ENABLED,
-    },
-    (isExtensionEnabled) => {
-      if (!isExtensionEnabled) {
-        return;
-      }
+  RQ.ConsoleLogger.setup();
+  RQ.RuleExecutionHandler.setup();
+  RQ.ScriptRuleHandler.setup();
+  RQ.SessionRecorder.setup();
+  RQ.UserAgentRuleHandler.setup();
+  RQ.RequestResponseRuleHandler.setup();
```

Comment:
@lazyvab I think it was fair to have all the handlers being initialized on the basis of extension is enabled or not. Do you think this is also causing enough delay for creating race conditions for AJAX interception?